packages/jest-cli/src/__tests__/watch.test.js

```diff
@@ -391,6 +396,119 @@ describe('Watch mode flows', () => {
     });
   });
 
+  it('allows WatchPlugins to modify only white-listed global config keys', async () => {
+    const pluginPath = `${__dirname}/__fixtures__/plugin_path_config_updater`;
+    const config = Object.assign({}, globalConfig, {
+      rootDir: __dirname,
+      watchPlugins: [pluginPath],
+    });
+
+    let currentOption;
+
+    jest.doMock(
+      pluginPath,
+      () =>
+        class WatchPlugin {
+          getUsageInfo() {
+            return {key: 'x', prompt: 'test option white-listing'};
+          }
+
+          run(globalConfig, updateConfigAndRun) {
+            updateConfigAndRun({[currentOption]: '__JUST_TRYING__'});
+            return Promise.resolve();
+          }
+        },
+      {virtual: true},
+    );
+
+    // updateGlobalConfig.mockReset();
+    const whiteListedOptions = [
```

Comment:
I think we could use `it.each([objectWithWhitelistedOptionsAndOtherOptions])('')`

https://facebook.github.io/jest/docs/en/api.html#testeachtable-name-fn

Reply:
Right! Good point…  Let me try…